### PR TITLE
Handle condition where `skip_spaces` could put the index out of range.

### PR DESCRIPTION
**Pull request by eahrold**
- ref: https://github.com/react-native-community/react-native-svg/issues/1083

### CODE REVIEW

**Comment by msqar** [comment]
Good job, man. I hope this gets merged asap.

**Comment by msqar** [comment]
@msand no mean to disrespect by tagging you but this PR fixes something really important, it's literally fixing this bug that breaks the library from being used. Hope we can merge it as soon as possible. Thanks in advance.

**Comment by msand** [comment]
Published v9.8.2
@msqar In the future, no need to rush these thing, you're able to use forks/specific commits without them being in the master branch / published to npm.

**Comment by msqar** [comment]
@msand oh thanks, didn't know. :) thanks for your time. And sorry to bother!

File: android/src/main/java/com/horcrux/svg/PathParser.java [hunk 1]
```diff
@@ -39,6 +39,10 @@ Path parse(String data) {
         while (i < l) {
             skip_spaces();
 
+            if (i >= l) {
+                break
```
Comment (msqar) [inline android/src/main/java/com/horcrux/svg/PathParser.java]: Was it complaining about the missing ";" when u fixed this? It is complaining on mine D:

**Comment by msqar** [comment]
@eahrold Oops, lint complaining about a missing ";" in the java file. Do you disable lint?

**Comment by eahrold** [comment]
@msqar you're absolutely right needs the `;`
My mistake. I ran the `npm run lint` but it doesn't catch native code.

```
> react-native-svg $ npm run lint

> react-native-svg@9.8.1 lint ../Code/react-native-svg
> eslint ./


../Code/react-native-svg/lib/extract/extractColor.js
  153:12  warning  'name' is defined but never used  no-unused-vars

../Code/react-native-svg/lib/util.js
  3:14  warning  'key' is defined but never used  no-unused-vars

✖ 2 problems (0 errors, 2 warnings)
```

**Comment by msand** [comment]
Those warnings are a bug in eslint. If the native code compiles it should be enough. Assumed you had done that.

**Comment by eahrold** [comment]
> Those warnings are a bug in eslint. If the native code compiles it should be enough. Assumed you had done that.

Unfortunately, I did not.  I threw the changes together late last night without even intending to do a PR, and they were so small I thought "why not", and then did the pr to develop.

@msand It was careless of me, and I apologize. I hope it didn't bum out your Sunday too much.

**Comment by msand** [comment]
No worries, takes seconds to cut a release anyway, and noone else seems to have experienced the issue, no hard feelings :) Thanks for the fix!

**Comment by msqar** [comment]
Thanks for the fix guys :)

**Comment by msand** [comment]
Btw, this was certainly on me, I should've added continous integration with unit and end to end tests long ago, or tried using the changes at the very least. I set up a e2e testing prototype this spring, and this reminds me I need finish that up.

**Comment by synoumar** [comment]
I am getting the same error of '**'Error while updating property 'd' of a View managed by: RNSVGPath'** while I create an array of String data

**Pull request merged**